src/basic/log.c

```diff
@@ -1045,16 +1071,24 @@ int log_struct_iovec_internal(
                 char header[LINE_MAX];
                 log_do_header(header, sizeof(header), level, error, file, line, func, NULL, NULL, NULL, NULL);
 
-                struct iovec iovec[1 + n_input_iovec*2];
-                iovec[0] = IOVEC_MAKE_STRING(header);
+                /* Each input iovec translates to 2 iovecs in our iovec array because we need to add a newline. */
+                if (n_input_iovec > IOVEC_MAX / 2)
```

Comment:
`if (n_input_iovec * 2 + 1 > IOVEC_MAX)`??

Reply:
I went for `n_input_iovec * 2 >= IOVEC_MAX)` which should be equivalent